app/src/testDeviceRelease/kotlin/de/digitalService/useID/coordinator/IdentificationCoordinatorTest.kt

```diff
@@ -96,1149 +99,2273 @@ class IdentificationCoordinatorTest {
 
-        job.cancel()
+        identificationCoordinator.cancelIdentification()
+        advanceUntilIdle()
+        Assertions.assertEquals(
+            SubCoordinatorState.Cancelled,
+            identificationCoordinator.stateFlow.value
+        )
+        verify(exactly = 1) { mockNavigator.popToRoot() }
+        verify(exactly = 3) { mockIdCardManager.cancelTask() }
+
+        scanJob.cancel()
     }
 
     @OptIn(ExperimentalCoroutinesApi::class)
     @Test
-    fun startIdentificationProcess_ScanErrorWithRedirect() = runTest {
-        val redirectUrl = "redirectUrl"
-
-        val testFlow: Flow<EidInteractionEvent> = flow {
-            emit(EidInteractionEvent.RequestCardInsertion)
-            emit(EidInteractionEvent.CardRecognized)
-            delay(500L)
-            throw IdCardInteractionException.ProcessFailed(ActivationResultCode.REDIRECT, redirectUrl, null)
-        }
+    fun errorProcessFailedScanStateReachedWithoutRedirectUrl() = runTest {
 
-        mockkStatic("android.util.Base64")
-        mockkStatic("android.net.Uri")
-        every { Base64.encodeToString(any(), any()) } returns redirectUrl
-        every { Uri.encode(any()) } returns redirectUrl
-        every { Uri.decode(any()) } returns redirectUrl
+        // SETUP
+        every { mockCoroutineContextProvider.IO } returns dispatcher
 
-        every { mockIDCardManager.identify(mockContext, testURL) } returns testFlow
+        val idCardManagerFlow = MutableStateFlow<EidInteractionEvent>(EidInteractionEvent.Idle)
+        every { mockIdCardManager.eidFlow } returns idCardManagerFlow
 
-        every { mockCoroutineContextProvider.IO } returns dispatcher
+        val mockIntent = mockkClass(Intent::class)
+        mockkConstructor(Intent::class)
+        every { anyConstructed<Intent>().addFlags(Intent.FLAG_ACTIVITY_NEW_TASK) } returns mockIntent
 
         val identificationCoordinator = IdentificationCoordinator(
-            mockContext,
-            mockAppCoordinator,
-            mockIDCardManager,
-            mockTrackerManager,
-            mockIssueTrackerManager,
-            mockCoroutineContextProvider
+            context = mockContext,
+            navigator = mockNavigator,
+            idCardManager = mockIdCardManager,
+            storageManager = mockStorageManager,
+            trackerManager = mockTrackerManager,
+            issueTrackerManager = mockIssueTrackerManager,
+            coroutineContextProvider = mockCoroutineContextProvider
         )
 
-        identificationCoordinator.startIdentificationProcess(testTokenURL, true)
+        var scanInProgress = false
+        val scanJob = identificationCoordinator.scanInProgress
+            .onEach { scanInProgress = it }
+            .launchIn(CoroutineScope(dispatcher))
 
-        verify { mockIDCardManager.cancelTask() }
+        val pinCallback = mockk<(String) -> Unit>()
+
+        val testRequestAuthenticationRequestConfirmation =
```

Comment:
This could also be an instance variable because we don't mutate it in the tests, do we? That way we have less redundancy and can use a more realistic request object by using more appropriate values than empty strings.